File: ios/Sources/EdgeFrameworks/Core/EdgeToolDefinition.swift
/// Provider-neutral metadata describing a tool and its expected arguments.
public struct EdgeToolDefinition: Hashable, Sendable {
    public let name: String
    public let description: String
    /// A JSON Schema string. The core stores it unchanged without validation.
    public let inputSchemaJSON: String

    public init(name: String, description: String, inputSchemaJSON: String) {
        self.name = name
        self.description = description
        self.inputSchemaJSON = inputSchemaJSON
    }
}

File: android/edge-frameworks-core/src/main/kotlin/io/github/rajibsarwar/edgeframeworks/EdgeToolDefinition.kt
package io.github.rajibsarwar.edgeframeworks

/**
 * Provider-neutral metadata describing a tool and its expected arguments.
 *
 * @property inputSchemaJson A JSON Schema string, stored unchanged without validation.
 */
data class EdgeToolDefinition(
    val name: String,
    val description: String,
    val inputSchemaJson: String
)
